internal/controller/dscinitialization/dscinitialization_controller.go

```diff
@@ -417,27 +440,24 @@ func (r *DSCInitializationReconciler) newMonitoringCR(ctx context.Context, dsci
 		},
 	}
 
-	if dsci.Spec.Monitoring.Metrics != nil {
-		defaultMonitoring.Spec.Metrics = dsci.Spec.Monitoring.Metrics
-	}
+	// Use CreateOrUpdate to handle both init creation and later DSCI update
+	_, err := ctrl.CreateOrUpdate(ctx, r.Client, defaultMonitoring, func() error {
+		defaultMonitoring.Spec.Metrics = &serviceApi.Metrics{}
```

Comment:
Why was resources.Apply() call replaced with this? is there any difference?

Reply:
Is metrics: {} a valid value ?